src/tests/ftest/osa/offline_drain.py

```diff
@@ -1,260 +1,282 @@
+
+    def test_osa_offline_drain_multiple_pools(self):
+        """Test ID: DAOS-7159.
+
+        Test Description: Validate Offline Drain with multiple pools
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=osa,osa_drain,offline_drain
+        :avocado: tags=OSAOfflineDrain,test_osa_offline_drain_multiple_pools
+        """
+        self.log.info("Offline Drain : Multiple Pools")
+        self.run_offline_drain_test(2, data=True)
+
+    def test_osa_offline_drain_during_rebuild(self):
+        """Test ID: DAOS-7159.
+
+        Test Description: Validate Offline Drain during rebuild
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=osa,osa_drain,offline_drain,rebuild
+        :avocado: tags=OSAOfflineDrain,test_osa_offline_drain_during_rebuild
+        """
+        self.test_during_rebuild = self.params.get("test_with_rebuild", "/run/rebuild/*")
+        self.log.info("Offline Drain : During Rebuild")
+        self.run_offline_drain_test(1, data=True)
+
+    def test_osa_offline_drain_after_snapshot(self):
+        """Test ID: DAOS-8057.
+
+        Test Description: Validate Offline Drain after taking snapshot.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=osa,osa_drain,offline_drain,checksum
+        :avocado: tags=OSAOfflineDrain,test_osa_offline_drain_after_snapshot
+        """
+        self.test_with_snapshot = self.params.get("test_with_snapshot", "/run/snapshot/*")
+        self.log.info("Offline Drain : After taking snapshot")
+        self.run_offline_drain_test(1, data=True)
+
+    def test_osa_offline_drain_with_less_pool_space(self):
+        """Test ID: DAOS-7160.
+
+        Test Description: Drain rank after with less pool space.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=osa,osa_drain,offline_drain,offline_drain_full
+        :avocado: tags=OSAOfflineDrain,test_osa_offline_drain_with_less_pool_space
+        """
+        self.log.info("Offline Drain : Test with less pool space")
+        oclass = self.params.get("pool_test_oclass", '/run/pool_capacity/*')
+        pool_fillup = self.params.get("pool_fillup", '/run/pool_capacity/*')
+        self.run_offline_drain_test(1, data=True, oclass=oclass, pool_fillup=pool_fillup)
+
+    def test_osa_offline_drain_with_multiple_ranks(self):
+        """Test ID: DAOS-4753.
+
+        Test Description: Drain multiple ranks at the same time.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=osa,osa_drain,offline_drain,offline_drain_full
+        :avocado: tags=OSAOfflineDrain,test_osa_offline_drain_with_multiple_ranks
+        """
+        self.log.info("Offline Drain : Test with mutiple ranks")
+        self.test_with_multiple_ranks = self.params.get("test_with_multiple_ranks",

```

Comment:
We really have to stop setting class variables like this and instead pass as arguments to the appropriate functions. Doing this makes it extremely difficult to debug/refactor/make changes.

Reply:
Sure. will definitely consider this change.